source/slang/slang-check-conversion.cpp

```diff
@@ -547,8 +673,27 @@ namespace Slang
            !canCoerce(toType, fromInitializerListExpr->type, nullptr))
             return _failedCoercion(toType, outToExpr, fromInitializerListExpr);
 
+        // try to invoke the user-defined constructor if it exists
+        if(_invokeExprForExplicitCtor(toType, fromInitializerListExpr, outToExpr))
+        {
+            return true;
+        }
+
+        // try to invoke the synthesized constructor if it exists
+        if (!_invokeExprForSynthesizedCtor(toType, fromInitializerListExpr, outToExpr))
+        {
+            // TODO: check if it's C-style initializer list
+            // if it's not, return error.
+            // if it is, return, fall-back to legacy logic
+            getSink()->diagnoseRaw(Severity::Error, "Synthesized Constructor is not found\n");
```

Comment:
should turn into a proper diagnostic, instead of `diagnoseRaw`.

Reply:
Definitely, it's changed in the later commit.